src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -108,14 +109,59 @@ static void uct_cuda_ipc_cache_invalidate_regions(uct_cuda_ipc_cache_t *cache,
               cache->name, from, to);
 }
 
-ucs_status_t uct_cuda_ipc_unmap_memhandle(void *rem_cache, uintptr_t d_bptr,
+static
+ucs_status_t uct_cuda_ipc_get_rem_cache(uct_cuda_ipc_iface_t *iface, pid_t pid,
+                                        uct_cuda_ipc_cache_t **cache)
+{
+    ucs_status_t status = UCS_OK;
+    char target_name[64];
+    khiter_t khiter;
+    int khret;
+
+    ucs_recursive_spin_lock(&iface->remote_cache.lock);
+
+    khiter = kh_put(cuda_ipc_rem_cache, &iface->remote_cache.hash, pid,
+                    &khret);
+    if ((khret == UCS_KH_PUT_BUCKET_EMPTY) ||
+        (khret == UCS_KH_PUT_BUCKET_CLEAR)) {
+        snprintf(target_name, sizeof(target_name), "dest:%d", pid);
+        status = uct_cuda_ipc_create_cache(cache, target_name);
+        if (status != UCS_OK) {
+            ucs_error("could not create create cuda ipc cache: %s for pid %d",
+                      ucs_status_string(status), pid);
+            goto err;
```

Comment:
error handling seems to be incorrect. If you print error and return here, the following call to `uct_cuda_ipc_get_rem_cache` with the same pid will find the uninitialized value in the hash.
Probably need to delete the element here or use `kh_get` instead

Reply:
Thanks @brminich